pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -639,6 +639,15 @@ func (vs *VSphere) AttachDisk(vmDiskPath string, storagePolicyID string, nodeNam
 	}
 	requestTime := time.Now()
 	diskUUID, err = attachDiskInternal(vmDiskPath, storagePolicyID, nodeName)
+	if err != nil {
+		if IsManagedObjectNotFoundError(err) {
```

Comment:
Can you test attach by invalidating the cache? You can do it by following these steps:
1. Create volume.
2. Move the nodes to different VCs under different folders etc. This should invalidate the cache.
3. Create a pod that uses the volume.

Reply:
Yes, I have tested this. By moving the VM to different VC and Datacenter

Reply:
As we discussed, this scenario is not tested that validates your code changes in AttachDisk. Please validate this scenario and update the PR description.